### ISSUE TITLE
Test reporter fails to build under 2.12.6

### ISSUE THREAD

**Issue by samschlegel** [bazelbuild/rules_scala#661]
Attempting to run a test under 2.12.6 appears to fail due to `-Xfatal-warnings` + unused pattern vars. I'm not sure why this wasn't caught with CI.
```
ERROR: /home/sam/.cache/bazel/_bazel_sam/d50f718107f06eb5b720bb473093cd22/external/io_bazel_rules_scala/scala/support/BUILD:3:1: scala @io_bazel_rules_scala//scala/support:test_reporter failed (Exit 1)
external/io_bazel_rules_scala/scala/support/JUnitXmlReporter.scala:62: warning: pattern var e in method apply is never used; `e@_' suppresses this warning
      case e: RunCompleted =>
           ^
...
external/io_bazel_rules_scala/scala/support/JUnitXmlReporter.scala:458: warning: pattern var e in lazy value hostname is never used; `e@_' suppresses this warning
      case e: UnknownHostException => "unknown"
           ^
error: No warnings can be incurred under -Xfatal-warnings.
19 warnings found
one error found
19 warnings found
one error found
java.lang.RuntimeException: Build failed
        at io.bazel.rulesscala.scalac.ScalacProcessor.compileScalaSources(ScalacProcessor.java:242)
        at io.bazel.rulesscala.scalac.ScalacProcessor.processRequest(ScalacProcessor.java:67)
        at io.bazel.rulesscala.worker.GenericWorker.runPersistentWorker(GenericWorker.java:45)
        at io.bazel.rulesscala.worker.GenericWorker.run(GenericWorker.java:111)
        at io.bazel.rulesscala.scalac.ScalaCInvoker.main(ScalaCInvoker.java:41)
```

Will follow up shortly with a PR that does as suggested, but I'm very curious as to why this isn't caught in CI
